comit/src/ledger/bitcoin.rs

```diff
@@ -0,0 +1,114 @@
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::fmt;
+
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    strum_macros::Display,
+    strum_macros::EnumString,
+)]
+#[serde(rename_all = "lowercase")]
+#[strum(serialize_all = "lowercase")]
+pub enum Bitcoin {
+    Mainnet,
+    Testnet,
+    Regtest,
+}
+
+impl From<Bitcoin> for ::bitcoin::Network {
+    fn from(bitcoin: Bitcoin) -> ::bitcoin::Network {
+        match bitcoin {
+            Bitcoin::Mainnet => ::bitcoin::Network::Bitcoin,
+            Bitcoin::Testnet => ::bitcoin::Network::Testnet,
+            Bitcoin::Regtest => ::bitcoin::Network::Regtest,
+        }
+    }
+}
+
+impl From<::bitcoin::Network> for Bitcoin {
+    fn from(network: ::bitcoin::Network) -> Self {
+        match network {
+            bitcoin::Network::Bitcoin => Bitcoin::Mainnet,
+            bitcoin::Network::Testnet => Bitcoin::Testnet,
+            bitcoin::Network::Regtest => Bitcoin::Regtest,
+        }
+    }
+}
+
+/// Module specifically desgined for use with the `serde(with)` attribute.
```

Comment:
Technically, this is not a module but just a single function. Hence, it is also not gonna work with the `serde(with)` attribute :)

Two options:

1. Fix the documentation
2. Make it a module and provide also provide the `serialize` variant.